src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -337,14 +296,14 @@ private Lesson parseNewLessonDetails(String input) throws CommandException {
         String timeAndDay = allDetails[INDEX_DAY_TIME];
 
         String link = allDetails[INDEX_LINK];
-        if (!isValidLink(link) && !link.equals(EMPTY_STRING)) {
+        if (!Lesson.isValidLink(link) && !link.equals(EMPTY_STRING)) {
```

Comment:
Good job moving validation to the respective class. Makes it more organized and logical. Same for email validation (isValidEmail).